core/domain/state_domain.py

```diff
@@ -917,6 +917,26 @@ def delete_content_id_for_translation(self, content_id):
         else:
             self.translations_mapping.pop(content_id, None)
 
+    def get_translation_counts(self):
+        """Return a dict representing the number of translation available in a
+        languages in which there exist at least one translation in the
+        WrittenTranslation object.
+
+        Returns:
+            dict(str, int). A dict with language code as a key and number of
+            translation available in that language as the value.
+        """
+        translation_counts = {}
```

Comment:
Consider using collections.defaultdict to simplify the logic (if you use that, you don't need to have the if/else branch, and you can convert to a regular dict at the end).